src/io.c

```diff
@@ -1872,26 +1870,57 @@ static Obj FuncINPUT_LINENUMBER(Obj self)
 static Obj FuncSET_PRINT_FORMATTING_STDOUT(Obj self, Obj val)
 {
     TypOutputFile * output = IO()->Output;
-    if (!output)
-        ErrorMayQuit("SET_PRINT_FORMATTING_STDOUT called while no output is opened\n", 0, 0);
-    while (output->prev)
+    while (output) {
+        if (!output->stream && output->file == 1)
+            output->format = (val != False);
+        output = output->prev;
+    }
+    return 0;
+}
+
+static Obj FuncPRINT_FORMATTING_STDOUT(Obj self)
+{
+    TypOutputFile * output = IO()->Output;
+    while (output) {
+        if (!output->stream && output->file == 1)
+            return output->format ? True : False;
         output = output->prev;
+    }
+    ErrorMayQuit("PRINT_FORMATTING_STDOUT called whiles stdout is not open", 0, 0);
+    return 0;
+}
+
+static Obj FuncSET_PRINT_FORMATTING_CURRENT(Obj self, Obj val)
+{
+    TypOutputFile * output = IO()->Output;
+    if (!output)
+        ErrorMayQuit("SET_PRINT_FORMATTING_CURRENT called while no output is open", 0, 0);
     output->format = (val != False);
-    return val;
+    if (output->stream)
+        CALL_2ARGS(SetPrintFormattingStatus, output->stream, val);
+    return 0;
+}
+
+static Obj FuncPRINT_FORMATTING_CURRENT(Obj self)
+{
+    TypOutputFile * output = IO()->Output;
+    if (!output)
```

Comment:
Do we have to test this, or should we assert it? Not sure if possible to (sensibly) get here. Either is fine.

Reply:
It can happen in HPC-GAP. In fact, we may need additional modifications there due to the way stdout is handled...